File: Tercero/Desarrollo Software-DS/Practica 1/Sesion2/Opcional/Prototipo/ClienteLite.rb
require './Trabajo.rb'

class ClienteLite < Cliente

    @@seguro_obra=false

    protected
    def self.clone()
      return ClienteLite.new()
    end

    public
    def self.initialize(n,e,dni)
        super(n,e,dni)
    end





    public
    def crearTrabajo(of, tipo, num,trbs)

            trb=Trabajo.new(of,tipo,num,self)
            puts "El cliente LITE #{@nombre} ha creado el trabajo con id #{num}"

            trbs.push(trb)

        return trbs
    end

    


end

File: Desarrollo Software-DS/Practica 1/Sesion2/Obligatoria/P1S2/src/p1s2/clienteLite.java
package p1s2;

public class clienteLite extends cliente {
    
    public static Boolean seguro_obra=false;
    
    public clienteLite(String n, int e, String dni) {
        super(n, e, dni);
    }
    
    @Override
     public void crearTrabajo(float of,String tipo,int num)
     {
           Boolean buscado=true; 
            for(int i=0 ;i<this.sistema.trabajosSistemas.size() && buscado ;i++)
           {
               if(this.sistema.trabajosSistemas.get(i).getnumeroTrabajo()==num)
               {
                   System.out.println("El trabajo ya existe en el sistema");
                   buscado=false;
               }
            }
            
            if(buscado==true){
                trabajo trb=new trabajo(of,tipo,num,this);
                System.out.println("El cliente LITE "+this.nombre+" ha creado el trabajo con id "+trb.num_trabajo);
                this.trabajosSubidos.add(trb);
                this.sistema.trabajosSistemas.add(trb);
                try {
                //Ponemos a "Dormir" el programa durante los s que queremos
                Thread.sleep(5*1000);
             } catch (Exception e) {
                System.out.println(e);
             }
            }
       
       }   
    
    
}